src/adapter/src/catalog.rs

```diff
@@ -1953,6 +1956,45 @@ pub enum DataSourceDesc {
     Progress,
 }
 
+impl DataSourceDesc {
+    /// Describes the ingestion to the adapter, which essentially just enriches the a higher-level
+    /// [`PlanIngestion`] with a [`ClusterId`].
+    pub fn ingestion(
```

Comment:
I got confused by the name thinking it was a getter. Maybe name this `from_ingestion` or `with_ingestion` to signal it's a constructor?

Reply:
Is it uncommon for an enum to have a function with one of its variant names that generates that variant? I'd expect `DataSourceDesc::ingestion` to return a `DataSourceDesc::Ingestion(..)`.

Reply:
gonna merge but lmk if you'd still like this changed--nbd to do it in post